core/distributed/preconditioner/schwarz.cpp

```diff
@@ -129,16 +170,35 @@ void Schwarz<ValueType, LocalIndexType, GlobalIndexType>::generate(
         GKO_INVALID_STATE(
             "Requires either a generated solver or an solver factory");
     }
+    auto dist_mat =
+        as<experimental::distributed::Matrix<ValueType, LocalIndexType,
+                                             GlobalIndexType>>(system_matrix);
 
     if (parameters_.local_solver) {
-        this->set_solver(gko::share(parameters_.local_solver->generate(
-            as<experimental::distributed::Matrix<
-                ValueType, LocalIndexType, GlobalIndexType>>(system_matrix)
-                ->get_local_matrix())));
-
+        this->set_solver(gko::share(
+            parameters_.local_solver->generate(dist_mat->get_local_matrix())));
     } else {
         this->set_solver(parameters_.generated_local_solver);
     }
+
+
+    if (parameters_.galerkin_ops && parameters_.coarse_solver) {
```

Comment:
maybe there should be an exception, if only one of those is set. I think that would be a configuration error, and the user should be notified.

Reply:
I think we need in general a better way to validate our configuration. If multiple factory parameters, then I think there should be a way to communicate that directly to the user rather than having the logic in the core file. Maybe we can think about adding some dependency support checks in the factory parameter setup agnostic to the concrete class.

Reply:
We could do that at a later point, but for now we should still throw if only one is set. I don't think it would be much of a burden here.

Reply:
Yes, sure.